DoePaAdminApp/ViewModel/DisplayAuftragsstatusViewModel.cs

```diff
@@ -0,0 +1,169 @@
+﻿using DoePaAdmin.ViewModel.Services;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using DoePaAdminDataModel.Stammdaten;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using OxyPlot;
+using OxyPlot.Axes;
+using OxyPlot.Series;
+using OxyPlot.Legends;
+using System.Collections.Generic;
+using Microsoft.VisualBasic;
+using DoePaAdminDataModel.Kostenrechnung;
+using DoePaAdminDataModel.DTO;
+
+namespace DoePaAdmin.ViewModel
+{
+    public class DisplayAuftragsstatusViewModel : DoePaAdminViewModelBase
+    {
+        private Geschaeftsjahr _selectedGeschaeftsjahr;
+        public Geschaeftsjahr SelectedGeschaeftsjahr
+        {
+            get => _selectedGeschaeftsjahr;
+            set => SetProperty(ref _selectedGeschaeftsjahr, value);
+        }
+
+        private ObservableCollection<Auftrag> _auftraege;
+        public ObservableCollection<Auftrag> Auftraege
+        {
+            get => _auftraege;
+            set => SetProperty(ref _auftraege, value);
+        }
+
+        public DisplayAuftragsstatusViewModel(IDoePaAdminService doePaAdminService) : base(doePaAdminService)
+        {
+            SelectedGeschaeftsjahr = Task.Run(async () => await DoePaAdminService
+                .GetGeschaeftsjahreAsync())
+                .Result
+                .FirstOrDefault(g => g.DatumVon <= DateTime.Now && g.DatumBis >= DateTime.Now);
+
+            Controller = new PlotController();
+            Controller.UnbindMouseDown(OxyMouseButton.Left);
+            Controller.BindMouseEnter(PlotCommands.HoverSnapTrack);
+
+            Charts = new();
+
+            Auftraege = new();
+
+            PropertyChanged += HandlePropertyChanged;
+        }
+
+        private void HandlePropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            switch (e.PropertyName)
+            {
+                case nameof(SelectedGeschaeftsjahr):
+                    Auftraege = new ObservableCollection<Auftrag>(SelectedGeschaeftsjahr.Auftraege);
+
+                    DrawCharts();
+
+                    break;
+            }
+        }
+
+        private async void DrawCharts()
```

Comment:
If it's an async method we should call it "DrawChartsAsync". And therefore also add a CancellationToken as parameter.